backoffice/src/Helpers/SearchHelpers/DroitQueryBuilder.cs

```diff
@@ -0,0 +1,372 @@
+                );
+            }
+            
+            if (!string.IsNullOrEmpty(form.LocationDescription))
+            {
+               query = query.Where(d => 
+                    d.LocationDescription != null && 
+                    (!string.IsNullOrEmpty(d.LocationDescription) &&
+                     (usePsql? EF.Functions.FuzzyStringMatchLevenshtein(form.LocationDescription.ToLower(), d.LocationDescription.ToLower()) :
+                         SearchHelper.GetLevenshteinDistance(form.LocationDescription.ToLower(), d.LocationDescription.ToLower())) < MaxLevenshteinDistance ||
+                     d.LocationDescription.ToLower().Contains(form.LocationDescription.ToLower()))
+                );
+            }
+
+
+            query = query.Where(
+                d => // long and lat to use location radius in calculation (method on droit data)
+                    ( form.InUkWaters == null || d.InUkWaters == form.InUkWaters ) &&
+                    ( form.RecoveredFromList.IsNullOrEmpty() ||
+                      ( d.RecoveredFrom.HasValue &&
+                        form.RecoveredFromList.Contains(d.RecoveredFrom.Value) ) )
+            );
+
+            //Wreck Material Filters
+            query = query.Where(d =>
+                form.IgnoreWreckMaterialSearch ||
+                d.WreckMaterials.Any(wm =>
+                    form.WreckMaterial != null &&
+                    ( !string.IsNullOrEmpty(wm.Description) &&
+                      ( usePsql
+                          ? EF.Functions.FuzzyStringMatchLevenshtein(
+                              form.WreckMaterial.ToLower(), wm.Description.ToLower())
+                          : SearchHelper.GetLevenshteinDistance(form.WreckMaterial.ToLower(),
+                              wm.Description.ToLower()) ) < MaxLevenshteinDistance ||
+                      wm.Description.ToLower().Contains(form.WreckMaterial.ToLower()) )) &&
+                d.WreckMaterials.Any(wm =>
+                    form.WreckMaterialOwner != null &&
+                    ( !string.IsNullOrEmpty(wm.Description) &&
+                      ( usePsql
+                          ? EF.Functions.FuzzyStringMatchLevenshtein(
+                              form.WreckMaterialOwner.ToLower(),
+                              wm.WreckMaterialOwner.ToLower())
+                          : SearchHelper.GetLevenshteinDistance(
+                              form.WreckMaterialOwner.ToLower(),
+                              wm.WreckMaterialOwner.ToLower()) ) < MaxLevenshteinDistance ||
+                      wm.WreckMaterialOwner.ToLower()
+                          .Contains(form.WreckMaterialOwner.ToLower()) )) &&
+                d.WreckMaterials.Any(wm =>
+                    form.ValueConfirmed == null || wm.ValueConfirmed == form.ValueConfirmed));
+            // the "in between" values here may need to be pull put to an if statement using the first conditional
+
+            if ( form.QuantityFrom != null )
```

Comment:
The any here is interesting with the breaking apart for the between 

WM 1 : 5
WM 2: 7

Form:

From: 6 to 6

Neither of them should catch because they're outside of the criteria, but because they're now checking any, they'll both hit as true because independently it does match.

I think the solution to this is a bigger query within the any that checks not null and does the between inside the any

Reply:
Worth having a test case for a similar scenario